compiler/rustc_passes/src/diagnostic_items.rs

```diff
@@ -100,6 +100,18 @@ fn collect<'tcx>(tcx: TyCtxt<'tcx>) -> FxHashMap<Symbol, DefId> {
 
     // Collect diagnostic items in this crate.
     tcx.hir().krate().visit_all_item_likes(&mut collector);
+    // FIXME(visit_all_item_likes): Foreign items are not visited
+    // here, so we have to manually look at them for now.
+    for foreign_module in tcx.foreign_modules(LOCAL_CRATE) {
```

Comment:
Ah I guess this is why using a diagnostic item didn't work when I tried it?

The rest of the PR LGTM (aside from the comment nit), but this here I cannot review I'm afraid. Cc @oli-obk